tests/test_viz.py

```diff
@@ -1,35 +1,34 @@
+import pytest
+from conftest import covmats  # noqa: F401
 import numpy as np
-from pyriemann.utils.viz import (plot_confusion_matrix, plot_embedding,
-                                 plot_cospectra)
 
+from pyriemann.utils.viz import (
+    plot_confusion_matrix,
+    plot_embedding,
+    plot_cospectra,
+)
+import importlib.util
 
-def generate_cov(Nt, Ne):
-    """Generate a set of cavariances matrices for test purpose."""
-    rs = np.random.RandomState(1234)
-    diags = 2.0 + 0.1 * rs.randn(Nt, Ne)
-    A = 2*rs.rand(Ne, Ne) - 1
-    A /= np.atleast_2d(np.sqrt(np.sum(A**2, 1))).T
-    covmats = np.empty((Nt, Ne, Ne))
-    for i in range(Nt):
-        covmats[i] = np.dot(np.dot(A, np.diag(diags[i])), A.T)
-    return covmats, diags, A
+plot = importlib.util.find_spec("matplotlib")
 
 
-def test_embedding():
+@pytest.mark.skipif(plot is None, reason="Matplotlib not installed")
```

Comment:
I would suggest to use such a function:

https://github.com/mne-tools/mne-python/blob/7636a58a744bb95b59816144cce7d01746fe6db7/mne/utils/_testing.py#L87